libs/libc/stdlib/lib_strtod.c

```diff
@@ -188,8 +227,17 @@ double strtod(FAR const char *str, FAR char **endptr)
 
   /* Process an exponent string */
 
-  if (*p == 'e' || *p == 'E')
+  if ((p10 == 10. && (*p == 'e' || *p == 'E'))
+     || (p10 == 16. && (*p == 'p' || *p == 'P')))
     {
+      /* if the Hexadecimal system */
+
+      if (p10 == 16.)
+        {
+          exponent *= 4;
+          p10 = 2.0;
```

Comment:
```suggestion
          p10 = 2.;
```
Just to follow the same style